Tests/Unit Tests/JSON Validator/TWTJSONObjectValidatorTestCase.m

```diff
@@ -0,0 +1,134 @@
+        for (NSDictionary *testValue in test[TWTTestKeywordTests]) {
+
+            if (![[self failingTests] containsObject:testValue[TWTTestKeywordDescription]]) {
+                BOOL shouldPass = [testValue[TWTTestKeywordValid] boolValue];
+                XCTAssertTrue([validator validateValue:testValue[TWTTestKeywordData] error:nil] == shouldPass,
+                              @"\nValue: %@\nSchema: %@\nshould have %@ed because %@. (%@)",
+                              testValue[TWTTestKeywordData], test[TWTTestKeywordSchema], shouldPass ? @"pass" : @"fail",
+                              testValue[TWTTestKeywordDescription], test[TWTTestKeywordDescription]);
+            }
+        }
+    }
+}
+
+
+- (void)testKnownFailingTests
+{
+    NSString *directoryPath = @"/Users/jillcohen/Developer/TWTValidation/Tests/JSONSchemaTestSuite/tests/draft4/";
+    for (NSDictionary *test in [self testsInDirectory:directoryPath]) {
+        TWTJSONObjectValidator *validator = [TWTJSONObjectValidator validatorWithJSONSchema:test[TWTTestKeywordSchema] error:nil warnings:nil];
+        for (NSDictionary *testValue in test[TWTTestKeywordTests]) {
+            if ([[self failingTests] containsObject:testValue[TWTTestKeywordDescription]]) {
+                BOOL shouldPass = [testValue[TWTTestKeywordValid] boolValue];
+                XCTAssertTrue([validator validateValue:testValue[TWTTestKeywordData] error:nil] == shouldPass, @"\nValue: %@\nSchema: %@\nshould have %@ed because %@. (%@)",
+                              testValue[TWTTestKeywordData], test[TWTTestKeywordSchema], shouldPass ? @"pass" : @"fail", testValue[TWTTestKeywordDescription], test[TWTTestKeywordDescription]);
+            }
+        }
+    }
+}
+
+
+
+- (void)testCustom
+{
+    NSString *directoryPath = @"/Users/jillcohen/Developer/TWTValidation/Tests/JSONSchemaCustom/";
+    for (NSDictionary *test in [self testsInDirectory:directoryPath]) {
+        TWTJSONObjectValidator *validator = [TWTJSONObjectValidator validatorWithJSONSchema:test[TWTTestKeywordSchema] error:nil warnings:nil];
+        for (NSDictionary *testValue in test[TWTTestKeywordTests]) {
+            BOOL shouldPass = [testValue[TWTTestKeywordValid] boolValue];
+            XCTAssertTrue([validator validateValue:testValue[TWTTestKeywordData] error:nil] == shouldPass, @"\nValue: %@\nSchema: %@\nshould have %@ed because %@. (%@)",
+                          testValue[TWTTestKeywordData], test[TWTTestKeywordSchema], shouldPass ? @"pass" : @"fail", testValue[TWTTestKeywordDescription], test[TWTTestKeywordDescription]);
+        }
+    }
+}
+
+
+- (NSArray *)testsInDirectory:(NSString *)directoryPath
+{
+    NSArray *testFiles = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:directoryPath error:nil];
+    NSMutableArray *tests = [[NSMutableArray alloc] init];
+
+    for (NSString *file in testFiles) {
+        if ([file containsString:@".json"]) {
```

Comment:
We probably just want to check the end, not the whole string, right? `[file hasSuffix:@".json"]`